### PR TITLE
DHIS2-5434: fix flashing chart when loading interpretation directly from URL

### PR DESCRIPTION
**Pull request by edoardo**
The PR also removes unnecessary analytics requests for the latest chart's data.

### CODE REVIEW

File: packages/app/src/reducers/visualization.js [hunk 1]
```diff
@@ -19,3 +19,5 @@ export default (state = DEFAULT_VISUALIZATION, action) => {
 // Selectors
 
 export const sGetVisualization = state => state.visualization;
+export const sGetInterpretations = state =>
+    state.visualization ? state.visualization.interpretations : [];
```
Comment (jenniferarnesen) [inline packages/app/src/reducers/visualization.js]: here you can use `sGetVisualization(state).interpretations : [];`

Comment (edoardo) [reply packages/app/src/reducers/visualization.js]: Hmm... default for visualization is {null}, and I think it was changed from `{}`.
I would leave that as it is.

File: packages/app/src/components/Visualization/__tests__/Visualization.spec.js [hunk 2]
```diff
@@ -247,11 +247,10 @@ describe('Visualization', () => {
         vis.instance().recreateChart = recreateChartFn;
 
         it('triggers a reflow when rightSidebarOpen prop changes', () => {
-            vis.setProps({
-                ...props,
-                current: undefined,
-                rightSidebarOpen: true,
-            });
+            // simulate prevProps ?!
```
Comment (jenniferarnesen) [inline packages/app/src/components/Visualization/__tests__/Visualization.spec.js]: would be great to get a test on the new logic

Comment (edoardo) [reply packages/app/src/components/Visualization/__tests__/Visualization.spec.js]: The tests have been updated to account for the new logic, in particular the ones for tDoLoadvisualization.

**Pull request merged**